chainer_chemistry/dataset/preprocessors/common.py

```diff
@@ -165,3 +168,83 @@ def construct_discrete_edge_matrix(mol, out_size=-1):
         adjs[ch, i, j] = 1.0
         adjs[ch, j, i] = 1.0
     return adjs
+
+def construct_supernode_feature(mol, atom_array, adjs, out_size=-1):
+    """
+    Construct an input feature x' for a supernode
+
+    Args:
+        mol (rdkit.Chem.Mol): Input molecule
+        atom_array (numpy.ndarray) : array of atoms
+        adjs (numpy.ndarray): N by N 2-way array, or |E| by N by N 3-way array where |E| is the number of edgetypes.
+        out_size (int): not used...
+
+    Returns:
+        super_node_x (numpy.ndarray); 1-way array, the supernode feature.
+        len(super_node_x) will be 2 + 2 + MAX_ATOMIC_NUM*2 for 2-way adjs, 2 + 4*2 + MAX_ATOMIC_NUM*2 for 3-way adjs
+
+    """
+
+    largest_atomic_number = MAX_ATOMIC_NUM
+
+    if mol is None:
+        raise MolFeatureExtractionError('mol is None')
+    N = mol.GetNumAtoms()
+    E = numpy.sum(adjs.flatten())
+    if E < 1.0:
+        E = 1.0
+
+    if out_size < 0:
+        size = N
+    elif out_size >= N:
+        size = out_size
+    else:
+        raise MolFeatureExtractionError('out_size {} is smaller than number '
+                                        'of atoms in mol {}'
+                                        .format(out_size, N))
+
+    # check the size of adjs
+    if adjs.ndim == 2:
+        super_node_x = numpy.zeros(2 + 2 + largest_atomic_number*2)
+    elif adjs.ndim == 3:
+        super_node_x = numpy.zeros(2 + 4*2 + largest_atomic_number*2)
+    else:
+        raise NotImplementedError('adjs.ndim should be 2 or 3')
```

Comment:
ValueError may be more suitable, (we can change it later).

Reply:
fixed.